src/SearchComponents/SimilarSearch.svelte

```diff
@@ -2,55 +2,72 @@
         data = await getResByIri(iri, $token);
         let linkedIri = '';
         resType = data["@type"].replace(ontology + ':', '');
-        for (const prop of props){
-            let linkedRes = data;
-            let linkStack = [];
-            let curr = prop["linkedVia"];
-            while(curr){
-                linkStack.push(curr['propName']);
-                curr = curr['linkedVia'];
-            }
-            linkedRes = data;
-            while (linkStack.length > 0){
-                curr = linkStack.pop();
-                linkedIri = linkedRes[ontology + ':' + curr + 'Value']['knora-api:linkValueHasTarget']['@id'];
-                linkedRes = await getResByIri(linkedIri, $token);
+        for (const prop of props) {
+            await getPropValue(prop, data);
+        }
+        await search();
+    }
+    async function getPropValue(prop, data, linkQueue = []){
+        if (prop.hasOwnProperty("linkResource")){
+            for (const linkProp of prop["linkResource"]["Props"]){
+                const linkedIri = data[ontology + ':' + prop["propName"] + 'Value']['knora-api:linkValueHasTarget']['@id'];
+                const newData = await getResByIri(linkedIri, $token);
+                await getPropValue(linkProp, newData, linkQueue.concat([prop["propName"]]));
             }
-            if(linkedRes.hasOwnProperty(ontology + ':' + prop['propName'])){
+        } else {
+            if(data.hasOwnProperty(ontology + ':' + prop['propName'])){
                 // TODO: Support all property types
-                switch(linkedRes[ontology + ':' + prop['propName']]["@type"]){
+                switch(data[ontology + ':' + prop['propName']]["@type"]){
                     case "knora-api:TextValue":
-                        prop["value"] = linkedRes[ontology + ':' + prop['propName']]["knora-api:valueAsString"];
+                        prop["value"] = data[ontology + ':' + prop['propName']]["knora-api:valueAsString"];
                         break;
                     case "knora-api:IntValue":
-                        prop["value"] = linkedRes[ontology + ':' + prop['propName']]["knora-api:intValueAsInt"].toString();
+                        prop["value"] = data[ontology + ':' + prop['propName']]["knora-api:intValueAsInt"].toString();
                         break;
                     default:
                         // TODO: Throw error
                         console.log("No suitable type found");
                         break;
                 }
-                prop["type"] = linkedRes[ontology + ':' + prop['propName']]["@type"];
+                prop["type"] = data[ontology + ':' + prop['propName']]["@type"];
+                if (linkQueue.length > 0) {
+                    prop["linkQueue"] = linkQueue;
+                }
                 searchInfo.push(prop);
+                console.log(searchInfo);
             }
-
         }
-        await search();
+    }
+    function getStringForProp(prop){
```

Comment:
For better readability it is preferable that a empty line should be inserted between functions

Reply:
Done